app/notifications/notifications_ses_callback.py

```diff
@@ -105,8 +105,12 @@ def determine_notification_bounce_type(notification_type, ses_message):
 
 
 def remove_emails_from_bounce(bounce_dict):
-    for recip in bounce_dict['bouncedRecipients']:
-        recip.pop('emailAddress')
+    if bounce_dict.get('mail').get('headers', None):
+        bounce_dict['mail'].pop('headers')
+    if bounce_dict.get('mail').get('commonHeaders'):
```

Comment:
I'd write that as 

```
if bounce_dict['mail'].get('headers', None):
    ....
```

So that we're more explicit that `mail` _must_ be there. In any case if it is not, this will fail anyway.

Reply:
If will fail either way, but I'll change it so it anyway to be clearer